sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/Helpers/FlowAnalysis/ControlFlowGraphTest.cs

```diff
@@ -1483,133 +1483,200 @@ public void Cfg_TryFinally()
             var blocks = cfg.Blocks.ToList();
 
             var tryStartBlock = blocks[0];
-            tryStartBlock.Should().BeOfType<SimpleBlock>();
-            tryStartBlock.Instructions.Should().HaveCount(2);
+            var tryEndBlock = blocks[1];
+            var finallyBlock = blocks[2];
+            var afterFinallyBlock = blocks[3];
 
-            var tryBodyBlock = blocks[1];
-            tryBodyBlock.Should().BeOfType<SimpleBlock>();
-            tryBodyBlock.Instructions.Should().HaveCount(2);
+            tryStartBlock.Should().BeOfType<BranchBlock>();
+            VerifyAllInstructions(tryStartBlock, "cw0", "cw0()");
 
-            var finallyBlock = blocks[2];
-            finallyBlock.Should().BeOfType<SimpleBlock>();
-            finallyBlock.Instructions.Should().HaveCount(2);
+            tryEndBlock.Should().BeOfType<BranchBlock>();
+            VerifyAllInstructions(tryEndBlock, "cw1", "cw1()");
+
+            finallyBlock.Should().BeOfType<BranchBlock>();
+            VerifyAllInstructions(finallyBlock, "cw2", "cw2()");
 
-            var afterFinallyBlock = blocks[3];
             afterFinallyBlock.Should().BeOfType<SimpleBlock>();
-            afterFinallyBlock.Instructions.Should().HaveCount(2);
+            VerifyAllInstructions(afterFinallyBlock, "cw3", "cw3()");
 
-            var exit = blocks.Last();
-            exit.Should().BeOfType<ExitBlock>();
+            blocks.Last().Should().BeOfType<ExitBlock>();
         }
 
         [TestMethod]
         [TestCategory("CFG")]
-        public void Cfg_TryFinally_Nested()
+        public void Cfg_TryCatchFinally_Return()
         {
             var cfg = Build(@"
             cw0();
             try
             {
                 cw1();
-                try
-                {
-                    cw2();
-                }
-                finally
-                {
-                    cw3();
-                }
+                return;
+            }
+            catch
+            {
+                cw2();
+                return;
             }
             finally
             {
-                cw4();
+                cw3(); // C# cannot return from finally
             }
-            cw5();");
+            cw4();");
 
             VerifyCfg(cfg, 7);
 
             var blocks = cfg.Blocks.ToList();
 
-            VerifyAllInstructions(blocks[0], "cw0", "cw0()");
-            VerifyAllInstructions(blocks[1], "cw1", "cw1()");
-            VerifyAllInstructions(blocks[2], "cw2", "cw2()");
-            VerifyAllInstructions(blocks[3], "cw3", "cw3()");
-            VerifyAllInstructions(blocks[4], "cw4", "cw4()");
-            VerifyAllInstructions(blocks[5], "cw5", "cw5()");
-            VerifyAllInstructions(blocks[6]);
+            var tryStartBlock = blocks[0];
+            var tryReturnBlock = blocks[1];

```

Comment:
Slightly confusing name. This is not a try return block. It's a try body that happens to have a return instruction.

I'd suggest it to:
var beforeTryBlock = blocks[0];
var tryBlock = blocks[1]; or tryBodyBlock